### PR TITLE
refactor: cleanup cdn samples

### PR DESCRIPTION
**Pull request by MonochromaticCoder**
## Description

b/280875956: add type hints
b/280880837: clean up

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: cdn/snippets.py [hunk 1]
```diff
@@ -23,199 +23,195 @@
...
 import hmac
-
-from six.moves import urllib
+from urllib.parse import parse_qs, urlsplit
 
 
 # [START cloudcdn_sign_url]
-def sign_url(url, key_name, base64_key, expiration_time):
+def sign_url(
+    url: str, key_name: str, base64_key: str, expiration_time: datetime
```
Comment (rsamborski) [inline cdn/snippets.py:34]: For readability please split into multiple lines ([see here for an example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/0ecdffa9b444aa01fab18324709cec9095c4bef1/compute/client_library/snippets/instances/create.py#L49-L55))

File: cdn/snippets.py [hunk 2]
```diff
@@ -23,199 +23,195 @@
...
-            separator='&' if query_params else '?',
-            expires=expiration_timestamp,
-            key_name=key_name)
+    digest = hmac.new(decoded_key, url_to_sign.encode("utf-8"), hashlib.sha1).digest()
+    signature = base64.urlsafe_b64encode(digest).decode("utf-8")
 
-    digest = hmac.new(
-        decoded_key, url_to_sign.encode('utf-8'), hashlib.sha1).digest()
-    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    return f"{url_to_sign}&Signature={signature}"
```
Comment (rsamborski) [inline cdn/snippets.py:61]: Please remove empty lines before the closure of the region tag if `black`  formatter doesn't enforce them.

File: cdn/snippets.py [hunk 3]
```diff
@@ -23,199 +23,195 @@
...
-            expires=expiration_timestamp,
-            key_name=key_name)
+    policy = f"URLPrefix={encoded_url_prefix}:Expires={expiration_timestamp}:KeyName={key_name}"
+
+    digest = hmac.new(decoded_key, policy.encode("utf-8"), hashlib.sha1).digest()
+    signature = base64.urlsafe_b64encode(digest).decode("utf-8")
+
+    signed_policy = f"Cloud-CDN-Cookie={policy}:Signature={signature}"
+
+    return signed_policy
```
Comment (rsamborski) [inline cdn/snippets.py:141]: ditto: please remove empty lines before the closure of the region tag if `black` formatter doesn't enforce them.

File: cdn/snippets.py [hunk 4]
```diff
@@ -23,199 +23,195 @@
...
-            key_name=key_name)
+    policy = f"URLPrefix={encoded_url_prefix}&Expires={expiration_timestamp}&KeyName={key_name}"
+
+    digest = hmac.new(decoded_key, policy.encode("utf-8"), hashlib.sha1).digest()
+    signature = base64.urlsafe_b64encode(digest).decode("utf-8")
 
-    digest = hmac.new(
-            decoded_key, policy.encode('utf-8'), hashlib.sha1).digest()
-    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    return f"{stripped_url}{'&' if query_params else '?'}{policy}&Signature={signature}"
```
Comment (rsamborski) [inline cdn/snippets.py:103]: ditto: please remove empty lines before the closure of the region tag if `black` formatter doesn't enforce them.

**Review by rsamborski — APPROVED** [review]
Nice work. Thanks. Approved with minor nits.

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**